implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java

```diff
@@ -148,6 +179,54 @@ private final SmallRyeHealth getHealth(Iterable<HealthCheck>... checks) {
             status = processChecks(additionalChecks, results, status);
         }
 
+        return createSmallRyeHealth(results, status);
+    }
+
+    @SafeVarargs
+    private final Uni<SmallRyeHealth> getHealthAsync(Iterable<AsyncHealthCheck>... checks) {
```

Comment:
Suggestion:

The produced Uni can be reused. Every time someone subscribes to that Uni, the checks will be called again.

Reply:
@cescoffier I don't understand your comment. Should I add this to JavaDoc or handle this somehow in the code?

Reply:
Once the Uni is created, you don't need to re-create it every time. Re-subscribing re-run it. 

```
AtomicInteger count = new AtomicInteger();
Uni<Integer> uni = Uni.createFrom().item(() -> count.getAndIncrement());

uni.await().indefinitely() -> 0
uni.await().indefinitely() -> 1
uni.await().indefinitely() -> 2
uni.await().indefinitely() -> 3
//... 
```

Reply:
ah, so you're suggesting to cache created Uni. But that won't work with dynamic registration as `additionalAsyncChecks` list.

Reply:
Yes, I realized that... Maybe we can have a cached uni for the "regular" case.